diff --git a/plat/freescale/imx8mq/gpc.c b/plat/freescale/imx8mq/gpc.c
index 0d04a0f..8333bdb 100644
--- a/plat/freescale/imx8mq/gpc.c
+++ b/plat/freescale/imx8mq/gpc.c
@@ -389,10 +389,10 @@ void imx_set_sys_lpm(bool retention)
 			 SLPCR_BYPASS_PMIC_READY | SLPCR_RBC_EN);
 
 		/* DDR enter retention */
-		ddrc_enter_retention();
+		//ddrc_enter_retention();
 	} else {
 		/* DDR exit retention */
-		ddrc_exit_retention();
+		//ddrc_exit_retention();
 	}
 
 	mmio_write_32(IMX_GPC_BASE + 0x14, val);
