diff --git a/plat/freescale/imx8mq/gpc.c b/plat/freescale/imx8mq/gpc.c
index 0d04a0fa..4e8f7bac 100644
--- a/plat/freescale/imx8mq/gpc.c
+++ b/plat/freescale/imx8mq/gpc.c
@@ -535,7 +535,7 @@ static void imx_gpc_set_affinity(uint32_t hwirq, unsigned cpu_idx)
 
 static void imx_gpc_pm_domain_enable(uint32_t domain_id, uint32_t on)
 {
-	uint32_t val;
+	uint32_t val, count;
 	uintptr_t reg;
 
 	/*
@@ -578,7 +578,8 @@ static void imx_gpc_pm_domain_enable(uint32_t domain_id, uint32_t on)
 	reg = IMX_GPC_BASE + (on ? 0xf8 : 0x104);
 	val = 1 << (domain_id > 3 ? (domain_id + 3) : domain_id);
 	mmio_write_32(reg, val);
-	while(mmio_read_32(reg) & val)
+	count = 100;
+	while((mmio_read_32(reg) & val) && count--)
 		;
 	imx_gpc_set_m_core_pgc(gpc_pu_m_core_offset[domain_id], false);
 }
