diff --git a/core/arch/arm/plat-imx/conf.mk b/core/arch/arm/plat-imx/conf.mk
index b05c7b6f..3bcbbc17 100644
--- a/core/arch/arm/plat-imx/conf.mk
+++ b/core/arch/arm/plat-imx/conf.mk
@@ -14,7 +14,7 @@ mx6sll-flavorlist = mx6sllevk
 mx7-flavorlist = mx7dsabresd
 mx7ulp-flavorlist = mx7ulpevk
 mx8m-flavorlist = mx8mqevk
-mx8mm-flavorlist = mx8mmevk
+mx8mm-flavorlist = mx8mmevk mx8mmsom
 
 ifneq (,$(filter $(PLATFORM_FLAVOR),$(mx6ul-flavorlist)))
 $(call force,CFG_MX6,y)
@@ -307,6 +307,10 @@ ifneq (,$(filter $(PLATFORM_FLAVOR),mx8mmevk))
 CFG_DDR_SIZE ?= 0x80000000
 endif
 
+ifneq (,$(filter $(PLATFORM_FLAVOR),mx8mmsom))
+CFG_DDR_SIZE ?= 0x80000000
+endif
+
 ifeq ($(filter y, $(CFG_PSCI_ARM32)), y)
 CFG_HWSUPP_MEM_PERM_WXN = n
 CFG_IMX_WDOG ?= y
diff --git a/core/arch/arm/plat-imx/tzasc.c b/core/arch/arm/plat-imx/tzasc.c
index 864f9ccd..c3774029 100644
--- a/core/arch/arm/plat-imx/tzasc.c
+++ b/core/arch/arm/plat-imx/tzasc.c
@@ -339,7 +339,7 @@ static int board_imx_tzasc_configure(vaddr_t addr)
 
 	return 0;
 }
-#elif defined(PLATFORM_FLAVOR_mx8mmevk)
+#elif defined(PLATFORM_FLAVOR_mx8mmevk) || defined(PLATFORM_FLAVOR_mx8mmsom)
 static int board_imx_tzasc_configure(vaddr_t addr)
 {
 	tzc_init(addr);
