--- a/opkg-build	2018-06-16 12:54:53.581191934 +0300
+++ b/opkg-build	2018-06-16 13:22:28.321938270 +0300
@@ -131,8 +131,8 @@ You probably want to chown these to a sy
 	disallowed_filename=`disallowed_field Filename`
 	[ "$?" -ne 0 ] && PKG_ERROR=1
 
-	if echo $pkg | grep '[^a-z0-9.+-]'; then
-		echo "*** Error: Package name $name contains illegal characters, (other than [a-z0-9.+-])" >&2
+	if echo $pkg | grep '[^a-z0-9.+-_]'; then
+		echo "*** Error: Package name $pkg contains illegal characters, (other than [a-z0-9.+-_])" >&2
 		PKG_ERROR=1;
 	fi
 
--- a/opkg-buildpackage	2017-01-30 22:52:45.000000000 +0300
+++ b/opkg-buildpackage	2018-06-16 13:23:29.985668241 +0300
@@ -65,8 +65,8 @@ pkg_appears_sane_control() {
 	required_field Maintainer >/dev/null
 	required_field Description >/dev/null
 
-	if echo $pkg | grep '[^a-z0-9.+-]'; then
-		echo "opkg-build: Error: Package name $name contains illegal characters, (other than [a-z0-9.+-])"
+	if echo $pkg | grep '[^a-z0-9.+-_]'; then
+		echo "opkg-build: Error: Package name $pkg contains illegal characters, (other than [a-z0-9.+-_])"
 		PKG_ERROR=1;
 	fi
 
